flow/scripts/detail_place.tcl

```diff
@@ -7,7 +7,15 @@ source $::env(PLATFORM_DIR)/setRC.tcl
 set_placement_padding -global \
     -left $::env(CELL_PAD_IN_SITES_DETAIL_PLACEMENT) \
     -right $::env(CELL_PAD_IN_SITES_DETAIL_PLACEMENT)
-detailed_placement
+if {[info exists env(WRITE_ON_FAIL)] && $::env(WRITE_ON_FAIL)} {
+  set result [catch {detailed_placement} errMsg]
+  if {$result != 0} {
+    write_db $::env(RESULTS_DIR)/3_5_place_dp.odb
```

Comment:
I disagree with the approach of writing the final odb file as if it is succeeded.  If you want to write 3_5_place_db-FAILED.odb that would make more sense to me.

Reply:
That does not work and defeats the purpose of this PR: Bazel must have the same artifact file name, even if placement failed. Checking for valid placement is a separate step afterwards. This separates the concern of producing an artifact to be used or inspected and checking for failure.

Reply:
If you write the success artifact how will make know that the step is actually failed?

Reply:
implemented via place.ok and route.ok files.